wallet/src/state.rs

```diff
@@ -42,6 +46,21 @@ impl ClientState {
         self.wallet.new_address(label)
     }
 
+    /// Generate a new transaction.
+    pub fn new_transaction<R: RngCore + CryptoRng>(
```

Comment:
Right now this is a stub function, but going forward, do we want to have a method like this, vs exposing a transaction builder? (Trying to imagine what this looks like if it had all of the arguments necessary to specify transaction details).

Reply:
I was thinking that we'd want the transaction building to happen in a method on the `ClientState` just because that's where we have knowledge of e.g. which notes we have (https://github.com/penumbra-zone/penumbra/blob/main/crypto/src/transaction/builder.rs#L40). So from the `pcli` side we'd pass in the data from the user's request "I want X of asset Y to go to address Z", and then the this method tries to build a transaction, bailing out if e.g. there are insufficient notes to send X of asset Y.